presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -199,14 +199,14 @@ public SemiTransactionalHiveMetastore(HdfsEnvironment hdfsEnvironment, ExtendedH
             }
         }
 
-        Optional<Map<String, Map<String, HiveColumnStatistics>>> delegateResult = delegate.getPartitionColumnStatistics(databaseName, tableName, partitionNamesToQuery.build(), columnNames);
-        if (delegateResult.isPresent()) {
-            resultBuilder.putAll(delegateResult.get());
+        Map<String, Map<String, HiveColumnStatistics>> delegateResult = delegate.getPartitionColumnStatistics(databaseName, tableName, partitionNamesToQuery.build(), columnNames);
+        if (!delegateResult.isEmpty()) {
```

Comment:
`delegateResult.isPresent()` signalled impossibility of calculating stats (see `com.facebook.presto.hive.metastore.SemiTransactionalHiveMetastore#getTableColumnStatistics` exit path for certain operations). Empty map might be understood as "no partitions" and so, this is a different case.

Reply:
`delegateResult.isPresent() signalled impossibility of calculating stats` - you mean impossibility of getting stats? And `impossibility of getting stats` means that they were not computed? If they were not computed for a give partition why do we need an Optional to identify that?